### DISCUSSION TITLE
Create a video element

### DISCUSSION THREAD

**Discussion by CarrieEBeaton** [katspaugh/wavesurfer.js#3339] [Q&A] [answered]
I created this WaveSurfer element 

 this.wavesurfer = WaveSurfer.create({
      container: '#waveform',
      waveColor: 'rgb(200, 0, 200)',
      progressColor: 'rgb(100, 0, 100)',
      height: 100,
      backend: 'MediaElement',
      normalize: true,
      media: document.querySelector('video'),
    });

Added this to the HTML to display the video 
<div id="waveform"></div>

And I have this video element to display the video:

<video #video
       [ngClass]="classes"
       preload="auto"
       webkit-playsinline
       playsinline
       [src]="src">
</video>

According to the documentation that should be all I need to have my WaveSurfer show up and be able to play/pause my video w/ the WaveSurfer component.  Unfortunately, the wave are not rendering on the Canvas unless I add waveSurver.load(audioUrl) and when I do that it causes console errors for the video.  It's odd because I can click on the blank Canvas and get the video to play and the element is in the DOM, it's just not rendering properly.

The wave surfer does not appear to be in a ready state because,

 this.wavesurfer.on('ready', () => { console.log('am I ready?'});  is not getting called unless I add the load method.

Any suggestion?  What am I missing.

Thank you

**Comment by CarrieEBeaton** [comment]
I also added width 100% but that didn't make the wavebar show up.

**Comment by CarrieEBeaton** [comment]
This is the error I get when I add the load w/ the audio url.  zone.js:1681 Uncaught Error: this._notifyBufferUpdated is not a function, TypeError: this._notifyBufferUpdated is not a function

**Comment by katspaugh** [comment, marked as answer]
You might be creating wavesurfer before Angular (?) has set the src attribute.

**Reply by CarrieEBeaton** [reply]
That helped with getting the wavebar to render correctly and sync; however, I still see this error in the console - This is the error I get when I add the load w/ the audio url. zone.js:1681 Uncaught Error: this._notifyBufferUpdated is not a function, TypeError: this._notifyBufferUpdated is not a function.  I was hoping by rendering it w/ the media and src attribute instead of using the load method that it would get rid of the error in the console.  Any ideas on the console error issue?  I really appreciate your help.  Thank you in advance.

**Reply by katspaugh** [reply]
That error isn’t from wavesurfer
